internal/apiserver/storage.go

```diff
@@ -114,7 +113,6 @@ func NewStorageAPIServer(db *pgxpool.Pool, nc *nats.Conn, certFile, keyFile stri
 
 	databaseChecker := newDatabaseChecker(db, logger)
 	serverConfig.AddReadyzChecks(databaseChecker)
-	serverConfig.AddHealthChecks(healthz.PingHealthz)
```

Comment:
Do you have any specific reason to remove this health check?

Reply:
As indicated in the commit message, this is already included by default. Adding it a second time may introduce race conditions that lead to storage failures (depending on the goroutine start timing). This was causing the e2e tests to fail sometimes, introducing flakiness.